### ISSUE TITLE
Add `projects::Toggle` command

### ISSUE THREAD

**Issue by neunato** [zed-industries/zed#21789]
### Check for existing issues

- [X] Completed

### Describe the feature

This would make it possible to use a single keybind for opening _and_ closing the projects menu, and make it consistent with other pickers as we already have `command_palette::Toggle`, `file_finder::Toggle`, `language_selector::Toggle`

### Environment

Zed: v0.164.2 (Zed)
OS: Linux X11 arch unknown
Memory: 62.5 GiB
Architecture: x86_64
GPU: NVIDIA GeForce RTX 2060 with Max-Q Design || NVIDIA || 565.57.01

**Comment by HarshNarayanJha** [comment]
Can you clarify what do you mean by project's menu, maybe by screenshots?

Are you talking about the `projects::OpenRecent` action?

**Comment by neunato** [comment]
Yeah, I'm talking about the project picker, what you get by firing up `projects::OpenRecent` or `projects::OpenRemote`.

**Comment by neunato** [comment]
Maybe introducing `projects::ToggleRecent` and `projects::ToggleRemote` instead would work better.

**Comment by ankddev** [comment]
I can work on this.

**Comment by RPG-Alex** [comment]
> Maybe introducing `projects::ToggleRecent` and `projects::ToggleRemote` instead would work better.

Can you show a screenshot of what you mean for this? I am interested in taking this issue but want to make sure I understand what you want added for the command(s)?

**Comment by neunato** [comment]
Simply put, `projects::ToggleRecent` would show the recent projects panel if it's not already shown, and hide it if it is. The same applies for `projects::ToggleRemote`.

**Comment by sofaspawn** [comment]
Hey @RPG-Alex , I was curious if you're still working on this issue or is it okay for me to join in?

**Comment by RPG-Alex** [comment]
@sofaspawn hey was going to focus on this after the holidays. Welcome to start on it. Can work together as well.

**Comment by RPG-Alex** [comment]
Hey @sofaspawn I will email you, lets collaborate on this issue

**Comment by RPG-Alex** [comment]
@neunato I was inspecting the editor and how is what you want different than the default: `Ctrl+Alt+o` that opens the recent projects menu, and `Ctrl+Shift+Alt+o`? Sorry if I am misunderstanding but it seems to be built into Zed already.

**Comment by neunato** [comment]
> This would make it possible to use a single keybind for opening _and_ closing the projects menu

Open = press twice remains open
Toggle = press twice must close

**Comment by kaf-lamed-beyt** [comment]
> Yeah, I'm talking about the project picker, what you get by firing up `projects::OpenRecent` or `projects::OpenRemote`.

@neunato, just as @RPG-Alex, mentioned. I think this is already built into Zed. doing `Ctrl+Alt+O` opens the project menu for ya, listing the recent ones. Perhaps, you could elaborate on what you want `project::Toggle` to do.

**Comment by TeenCoder159** [comment]
Do you just mean closing and opening the file structure panel? Because that feature is already there

**Comment by RPG-Alex** [comment]
Hey I am not going to work on this issue, some one else is welcome to it.

**Comment by TeenCoder159** [comment]
I think you want to try pressing `cmd + option + o` or `ctrl+alt+o` or `cmd+ctrl+o` ?

**Comment by xtanion** [comment]
Hi, can I work on this?

**Comment by sir-geronimo** [comment]
When using the shortcut to "Open Recent" (Command ⌘ + Option ⌥ + O), or using the `File > Open Recent...` `MenuItem`, it cycles between the items on the list; but when clicking on the current window's name, it just opens the panel indefinitely.

What is the expected behaviour of this `projects: open recent` command?

In case the behaviour is changed from "open" to "toggle", should it keep the "cycle items" behaviour when using the shortcut or menu item? 🤔

**Comment by FloppyDisco** [comment]
hey might be worth noting that if the desired behavior is opening RecentProjects and then being able to close it again with the same keybinding this is already doable with `"context"`

```
  {
    "bindings": {
      "cmd-o": "projects::OpenRecent"
    }
  },
  {
    "context": "RecentProjects > Editor",
    "bindings": {
      "cmd-o": "menu::Cancel"
    }
  }
```
